Mathlib/CategoryTheory/Limits/IsConnected.lean

```diff
@@ -137,4 +137,49 @@ theorem isConnected_iff_of_initial (F : C ⥤ D) [F.Initial] : IsConnected C ↔
 
 end Functor
 
+section
+
+variable (C : Type*) [Category C]
+
+-- note : it seems making the following two as instances breaks things, so these are lemmas.
+lemma isConnected_of_hasInitial [Limits.HasInitial C] : IsConnected C := by
+  letI : Nonempty C := ⟨⊥_ C⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [⊥_ C, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.symm <| Zag.of_hom <| Limits.initial.to _, Zag.of_hom <| Limits.initial.to _⟩
+
+lemma isConnected_of_hasTerminal [Limits.HasTerminal C] : IsConnected C := by
+  letI : Nonempty C := ⟨⊤_ C⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [⊤_ C, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.of_hom <| Limits.terminal.from _, Zag.symm <| Zag.of_hom <| Limits.terminal.from _⟩
+
+/-- Prove that a category is connected by supplying an explicit initial object. -/
+lemma isConnected_of_isInitial {x : C} (h : Limits.IsInitial x) : IsConnected C := by
+  letI : Nonempty C := ⟨x⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [x, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.symm <| Zag.of_hom <| h.to _, Zag.of_hom <| h.to _⟩
+
+/-- Prove that a category is connected by supplying an explicit terminal object. -/
+lemma isConnected_of_isTerminal {x : C} (h : Limits.IsTerminal x) : IsConnected C := by
+  letI : Nonempty C := ⟨x⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [x, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.of_hom <| h.from _, Zag.symm <| Zag.of_hom <| h.from _⟩
```

Comment:
Let's prove these two first, then the `Has` versions should just be one line using [`terminalIsTerminal`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Limits/Shapes/Terminal.html#CategoryTheory.Limits.terminalIsTerminal)

Reply:
right, I had put the `has` version first when coding it, but it make much more sense!